### ISSUE TITLE
📎 Organize components/pages into features

### ISSUE THREAD

**Issue by sruenwg** [sugar-cat7/vspo-portal#262]
### Description

Discuss and implement the reorganization of the frontend code into features.

We concluded in https://github.com/sugar-cat7/vspo-portal/pull/259 that we should consider organizing the frontend by feature, following the `bulletproof-react` project structure.
The main motivation for this is to organize component/page utilities such that they live close to and are scoped only to the components/pages in which they are meant to be used (as opposed to the current design where utilities must either be globally scoped or file-scoped).

Potential points to discuss:
1. What comprises a *feature*? What features does the app have?
2. Which components and pages might belong to which feature?
3. Which components are generic enough to remain in `/components`? Which components might make sense to be refactored into a more generic component which can then live in `/components`?
4. Is there anything aside from the `/features` directory which would make sense to adopt from the `bulletproof-react` structure?

**Comment by sugar-cat7** [comment]
Certainly, it's incorrect to say that features and pages have a one-to-one relationship; it seems more appropriate to group related files based on modules or functional units, such as grouping all clip-related functionality together.

1. Personally, I consider the following to be the unique features of the current web application:

   - Broadcasting information posting (archives, upcoming, live, free chat (though it's debatable whether this belongs here))
   - Clip posting (YouTube, Twitch)
   - Event posting (List, Detail)
   - Announcements posting (Site News)
   - Posting site usage information (About?)
   - Terms posting (Terms/Privacy Policy)

Therefore, I don't feel any discomfort with the structure @sruenwg has shown in point 2.

Regarding point 3, I also think it would be good to generalize LivestreamDetailsModal, LivestreamCard into components like Modal and Card, and then customize them in features like ClipCard for clips and LivestreamCard for livestreams. It also seems appropriate to encapsulate specific fetcher functions from lib/api.ts and type definitions from types within their respective features.

I don't have anything particular for point 4. If I had to mention something, it would be about splitting the util files. But even that seems to be a matter of transferring specific logic and making adjustments if the readability is poor.
